### PR TITLE
feat: add thrift

### PR DESCRIPTION
**Pull request by amaanq**
This PR adds a thrift parser+queries

cc: @leaxoy since they first brought up thrift in #2282

I'm a little iffy on the queries, so I'd like a few pairs of eyes to review that, but the actual grammar [here](https://github.com/duskmoon314/tree-sitter-thrift/blob/main/grammar.js) is very solid imo and is extensively tested.

Thanks!
-Amaan

### CODE REVIEW

**Comment by amaanq** [comment]
Force pushed due to typo

**Comment by leaxoy** [comment]
Good job!!

**Comment by amaanq** [comment]
Force pushed again to bump revision

**Comment by amaanq** [comment]
> Good job!!

Thanks! :)

File: queries/thrift/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  "typedef"
+] @type.definition
+
+[
+  "const"
+] @type.qualifier
+
+[
+  "*"
+] @character.special
```
Comment (ObserverOfTime) [inline queries/thrift/highlights.scm:122]:
```suggestion
[
  "*"
] @punctuation.special
```

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: queries/thrift/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+(namespace) @namespace
+
+; Includes
+
+[
+  "include"
+  "cpp_include"
+] @include
+
+(include_path) @definition.import
```
[the remark this answers is not in the document]
Comment (amaanq) [reply queries/thrift/highlights.scm:12]: is there no way to make an include path a special highlight? would be nice

File: queries/thrift/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+(number) @number
+
+(double) @float
+
+(boolean) @boolean
+
+[
+  "typedef"
+] @type.definition
```
[the remark this answers is not in the document]
Comment (amaanq) [reply queries/thrift/highlights.scm:114]: should I remove it or have a definition like

```js
typedef: () => token("typedef")
```
```scm
(typedef) @type.definition
```

..?

**Comment by amaanq** [comment]
@theHamsta mind if I request for you to review the changes suggested and answer my questions about them? :smile:

File: queries/thrift/highlights.scm (hunk 2 above)
[the remark this answers is not in the document]
Comment (ObserverOfTime) [reply queries/thrift/highlights.scm:12]: You can use something like `@string.include` and highlight it however you want locally.

File: queries/thrift/highlights.scm (hunk 3 above)
[the remark this answers is not in the document]
Comment (ObserverOfTime) [reply queries/thrift/highlights.scm:114]: In e.g. `typedef foo bar`, the `@type.definition` is `bar`.

[the remark this answers is not in the document]
Comment (amaanq) [reply queries/thrift/highlights.scm]: Ok, fixed that by aliasing identifier to typedef_definition in typedef, thanks!

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**